container-search/src/main/java/com/yahoo/search/rendering/JsonRenderer.java

```diff
@@ -787,5 +666,156 @@ void setGenerator(JsonGenerator generator) {
     void setTimeSource(LongSupplier timeSource) {
         this.timeSource = timeSource;
     }
-    
+
+    /**
+     * Received callbacks when fields of hits are encountered.
+     * This instance is reused for all hits of a Result since we are in a single-threaded context
+     * and want to limit object creation.
+     */
+    private static class FieldConsumer implements Hit.RawUtf8Consumer {
+
+        private final JsonGenerator generator;
+        private final boolean debugRendering;
+
+        private MutableBoolean hasFieldsField;
+
+        public FieldConsumer(JsonGenerator generator, boolean debugRendering) {
+            this.generator = generator;
+            this.debugRendering = debugRendering;
+        }
+
+        /**
+         * Called before each use of this for a hit to keep track of whether we
```

Comment:
the "setCurrent" is called at the start of rendering each hit, right?
and then "fieldsStart" is called for every field?

Reply:
Yes.